testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/problems/KnownProblemIds.groovy

```diff
@@ -58,14 +61,17 @@ class KnownProblemIds {
         'generic' : 'Generic'
     ]
 
-    private static final def KNOWN_DEFINITIONS = [
+    /**
+     * This map is used to validate that problems reported have known IDs, and display name.
+     * <p>
+     * Both the key and value is handled as a regular expression if the value is too dynamic.
+     */
+    private static final HashMap<String, String> KNOWN_DEFINITIONS = [
         'problems-api:missing-id' : 'Problem id must be specified',
         'problems-api:unsupported-additional-data' : 'Unsupported additional data type',
         'compilation:groovy-dsl:compilation-failed' : 'Groovy DSL script compilation problem',
-        'compilation:java:java-compilation-error' : 'Java compilation error',
-        'compilation:java:java-compilation-failed' : 'Java compilation error',
-        'compilation:java:java-compilation-warning' : 'Java compilation warning',
-        'compilation:java:java-compilation-advice' : 'Java compilation note',
+        // Flexible category, as the category id's last component, and the message, will be supplied by the compiler
```

Comment:
For the most part, I'd prefer exact content listed here. At the same time, if the list becomes unmanageable (dozens of tests fail and  the list of known IDs are unmanageable), then I'd switch to pattern matching. Is this the case here?

Reply:
I took a different approach here: I kept the literal message and only kept the end if IDs dynamic. I think this is a sweet spot for not making the list huge yet having confidence that the value incoming is a (probably) valid one.